packages/ember-toucan-core/src/components/form/controls/combobox.ts

```diff
@@ -0,0 +1,537 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { next } from '@ember/runloop';
+import { isEqual as emberIsEqual } from '@ember/utils';
+
+import { offset, size } from '@floating-ui/dom';
+
+import OptionComponent, {
+  selector as optionComponentSelector,
+} from '../../../-private/components/form/controls/combobox/option';
+import Chevron from '../../../-private/icons/chevron';
+
+import type { Middleware as VelcroMiddleware } from '@floating-ui/dom';
+import type { WithBoundArgs } from '@glint/template';
+
+export type Option = string | Record<string, unknown> | undefined;
+
+export interface ToucanFormComboboxControlComponentSignature<
+  OPTION extends Option
+> {
+  Args: {
+    /**
+     * A CSS class to add to this component's content container.
+     * Commonly used to specify a `z-index`.
+     */
+    contentClass?: string;
+
+    /**
+     * Sets the input to an errored-state via styling.
+     */
+    hasError?: boolean;
+
+    /**
+     * Sets the `disabled` attribute of the input.
+     */
+    isDisabled?: boolean;
+
+    /**
+     * Sets the `readonly` attribute of the input.
+     */
+    isReadOnly?: boolean;
+
+    /**
+     * A string to display when there are no results after the user's filter.
+     */
+    noResultsText?: string;
+
+    /**
+     * Called when the user makes a selection.
+     * It is called with the selected option (derived from `@options`) as its only argument.
+     */
+    onChange?: (option: unknown) => void;
+
+    /**
+     * The function called when a user types into the combobox textbox, typically used to write custom filtering logic.
+     */
+    onFilter?: (input: string) => OPTION[];
+
+    /**
+     * `@options` forms the content of this component.
+     *
+     * To support a variety of data shapes, `@options` is typed as `unknown[]` and treated as though it were opaque.
```

Comment:
AFAICT, it is typed as `Array<string | Record<string, unknown> | undefined>` though, not unknown! Which is good I think, but the docs are misleading here.

Reply:
Doh, nice catch.  This was left over from when we previously had `unknown[]` rather than `Array<string | Record<string, unknown> | undefined>`.  Fixed with commit: https://github.com/CrowdStrike/ember-toucan-core/pull/200/commits/45594a7f284e0d3e2105254365d3e2d0c3695fe9